src/containers/Tasks/Tasks.js

```diff
@@ -85,198 +83,197 @@ export /* istanbul ignore next */ class Tasks extends Component {
       };
     }
 
-    const { deleteError } = this.state;
     if (deleteError) {
       return {
-        clear: () => this.setState({ deleteError: null }),
+        clear: () => setDeleteError(null),
         error: deleteError
       };
     }
 
     return null;
   }
 
-  closeDeleteModal = () => {
-    this.setState({
-      showDeleteModal: false,
-      toBeDeleted: []
-    });
-  };
+  useEffect(() => {
+    document.title = getTitle({ page: 'Tasks' });
+  }, []);
+
+  useEffect(() => {
+    fetchData();
+  }, [JSON.stringify(filters), selectedNamespace]);
+
+  useWebSocketReconnected(fetchData, webSocketConnected);
 
-  deleteTask = task => {
-    const { name, namespace } = task.metadata;
-    deleteTask({ name, namespace }).catch(error => {
-      error.response.text().then(text => {
-        const statusCode = error.response.status;
+  function closeDeleteModal() {
+    setShowDeleteModal(false);
+    setToBeDeleted([]);
+  }
+
+  function handleDeleteTask(task) {
+    deleteTask({
+      name: task.metadata.name,
+      namespace: task.metadata.namespace
+    }).catch(err => {
+      err.response.text().then(text => {
+        const statusCode = err.response.status;
         let errorMessage = `error code ${statusCode}`;
         if (text) {
           errorMessage = `${text} (error code ${statusCode})`;
         }
-        this.setState({ deleteError: errorMessage });
+        setDeleteError(errorMessage);
       });
     });
-  };
+  }
 
-  handleDelete = async () => {
-    const { cancelSelection, toBeDeleted } = this.state;
-    const deletions = toBeDeleted.map(resource => this.deleteTask(resource));
-    this.closeDeleteModal();
+  async function handleDelete() {
+    const deletions = toBeDeleted.map(resource => handleDeleteTask(resource));
+    closeDeleteModal();
     await Promise.all(deletions);
     cancelSelection();
-  };
-
-  openDeleteModal = (selectedRows, cancelSelection) => {
-    const resourcesById = keyBy(this.props.tasks, 'metadata.uid');
-    const toBeDeleted = selectedRows.map(({ id }) => resourcesById[id]);
-    this.setState({ showDeleteModal: true, toBeDeleted, cancelSelection });
-  };
-
-  fetchData() {
-    const { filters, namespace } = this.props;
-    this.props.fetchTasks({ filters, namespace });
   }
 
-  render() {
-    const { loading, tasks, intl, namespace: selectedNamespace } = this.props;
-    const { showDeleteModal, toBeDeleted } = this.state;
+  function openDeleteModal(selectedRows, handleCancelSelection) {
+    const resourcesById = keyBy(tasks, 'metadata.uid');
+    setShowDeleteModal(true);
+    setToBeDeleted(selectedRows.map(({ id }) => resourcesById[id]));
+    setCancelSelection(() => handleCancelSelection);
```

Comment:
can just past the reference to the function here
`setCancelSelection(handleCancelSelection);`

Reply:
@AlanGreene I have been try your approach but I found the error:

<img width="1440" alt="Screen Shot 2564-05-10 at 19 55 27" src="https://user-images.githubusercontent.com/484530/117662393-b483a000-b1c9-11eb-9230-8575c8988e1f.png">

I've been try deleting task in `ClusterTasks` and found the same error also:

<img width="1440" alt="Screen Shot 2564-05-10 at 19 52 28" src="https://user-images.githubusercontent.com/484530/117663721-2d372c00-b1cb-11eb-976f-28af13ffd4d5.png">

I think it's because of `React.useState` expecting to return the next state if we pass a function to the first argument.  That's why I calls `setCancelSelection` with a function that return a callback. And the reason that I set the initial state to an empty function is because of we can prevent nullable check. 

This is reference that I read about `useState`: https://reactjs.org/docs/hooks-reference.html#functional-updates.

Please correct me if I'm wrong. ^^

Reply:
ah of course, you're right 👍 I forgot about the lazy initialisation with `useState`

Reply:
Do you want me to send a PR to fix in `ClusterTasks`?

Reply:
yes that would be great, thank you!

Reply:
The PR is here https://github.com/tektoncd/dashboard/pull/2026. ^^